crates/networking/p2p/kademlia.rs

```diff
@@ -254,6 +256,39 @@ impl KademliaTable {
 
         None
     }
+
+    /// Set the sender end of the channel between the kademlia table and the peer's active connection
+    /// This function should be called each time a connection is established so the backend can send requests to the peers
+    pub fn set_channels(&mut self, node_id: H512, channels: PeerChannels) {
+        let bucket_idx = bucket_number(self.local_node_id, node_id);
+        if let Some(peer) = self.buckets[bucket_idx]
```

Comment:
Access with brackets can panic, let's use [get](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get).


To avoid making this function too verbose, maybe use [map](https://doc.rust-lang.org/std/option/enum.Option.html#method.map)